vembrane/__init__.py

```diff
@@ -108,31 +108,41 @@ def parse_annotation_entry(entry: str,) -> List[str]:
     return list(map(str.strip, entry.split("|")))
 
 
-def eval_expression(
-    expression: str, idx: int, annotation: str, annotation_keys: List[str], env: dict,
-) -> bool:
-    env["ANN"] = Annotation(
-        idx,
-        dict(
-            map(
-                lambda v: ANN_TYPER.convert(v[0], v[1]),
-                zip(annotation_keys, parse_annotation_entry(annotation)),
-            )
-        ),
-    )
-    return eval(expression, globals_whitelist, env)
+class Expression:
+    def __init__(self, expression: str, ann_key: str = "ANN"):
+        self._expression = expression
+        self._ann_key = ann_key
+        self._has_ann = f"{ann_key}[" in expression
```

Comment:
atm we just check for the ann_key with "[" appended to determine if the expression actually contains an ANN[ access. However, someone might do something like `"foo" in "".join(ANN)` for whatever reason, in which case it'd fail